modules/core/src/lib/deck-renderer.js

```diff
@@ -0,0 +1,69 @@
+import log from '../utils/log';
+import LayersPass from '../passes/layers-pass';
+
+const LOG_PRIORITY_DRAW = 2;
+
+export default class DeckRenderer {
+  constructor(props) {
+    const {gl, layerManager, effectManager} = props;
+
+    this.gl = gl;
+    this.layerManager = layerManager;
```

Comment:
I would try to minimize coupling between `Manager` classes. The less they know about each other, the better.

Perhaps we could pass in the layer list to this manager in `draw` calls instead of keeping a reference to the `layerManager`?

Reply:
There are two reasons I am doing in this way
1. Besides the layers there are other data or handler we need from LayerManager class, would be messy if we get all of them in deck class and pass to deckRenderer.
2. I don't think DeckRenderer is a manager, cause it doesn't manage anything's life span, more like an action executor which retrieves data or resources from managers and takes action.